### PR TITLE
do not compress booleans to !0 and !1, do so at output

### PR DESCRIPTION
**Pull request by fabiosantoscode**
What do you think, @kzc? I think this should make evals a little bit faster, and save some memory in the process.

### CODE REVIEW

**Comment by kzc** [comment]
I am not in favor this PR. `OutputStream` should faithfully print the AST as is.

Further, this functionality is already possible with an existing compress option:

```
$ echo 'console.log(true, false);' | terser -c
console.log(!0,!1);

$ echo 'console.log(true, false);' | terser -c booleans=0
console.log(true,false);
```

**Comment by kzc** [comment]
People maintaining and debugging Terser are the primary users of the `beautify` output option. Everyone in the JS community else uses `prettier`. This PR would negatively impact debugging Terser and creating test cases.

File: lib/output.js [hunk 1]
```diff
@@ -1789,7 +1789,10 @@ function OutputStream(options) {
         output.print("super");
     });
     DEFPRINT(AST_Constant, function(self, output) {
-        output.print(self.getValue());
+        var is_beautify = output.option("beautify");
+        if (!is_beautify && self.getValue() === true) output.print("!1");
+        else if (!is_beautify && self.getValue() === false) output.print("!0");
+        else output.print(self.getValue());
```
Comment (kzc) [inline lib/output.js]: fwiw, the outputs are incorrect. `!1` is false, `!0` is true.
`self.getValue()` is called more than necessary.
The variable `is_beautify` is not needed.

Search old uglify issues and PRs for unanticipated problems associated with having the output not representing the AST. In particular: `Infinity` and `1 / 0`.

**Comment by rvanvelzen** [comment]
> I am not in favor this PR. `OutputStream` should faithfully print the AST as is.

Seconded. We've tried this before and it almost always produces more problems than it appears to solve.

**Comment by fabiosantoscode** [comment]
Closing

**Pull request closed without merging**